diff --git a/src/gnss.c b/src/gnss.c
index aeb73af..bed8a3d 100644
--- a/src/gnss.c
+++ b/src/gnss.c
@@ -27,6 +27,7 @@ gboolean gnss_upload_assistance_data(struct EG25Manager *manager)
         return FALSE;
     }
 
+#ifdef HAVE_MMGLIB
     /* ModemManager's Location is only available after unlocking */
     if(!manager->mm_location) {
         g_message ("Rescheduling upload since Location interface is not available, in %ds",
@@ -34,6 +35,7 @@ gboolean gnss_upload_assistance_data(struct EG25Manager *manager)
         manager->gnss_assistance_step = EG25_GNSS_STEP_LAST;
         return TRUE;
     }
+#endif
 
     manager->gnss_assistance_step = EG25_GNSS_STEP_FIRST;
     gnss_step(manager);
